Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -1035,9 +1042,9 @@ void DiscusMultipleScatteringCorrection::integrateCumulative(const DiscusData1D
                                                              const double xmax, std::vector<double> &resultX,
                                                              std::vector<double> &resultY,
                                                              const bool returnCumulative) {
+  assert(h.histogram().xMode() == HistogramData::Histogram::XMode::Points);
```

Comment:
I'm having a problem building this - I get an error on this assert line
```
4>C:\mantid-conda\mantid\Framework\Algorithms\src\DiscusMultipleScatteringCorrection.cpp(1045,3): error C2039: 'histogram': is not a member of 'Mantid::Algorithms::DiscusData1D'
4>C:\mantid-conda\mantid\Framework\Algorithms\inc\MantidAlgorithms/DiscusMultipleScatteringCorrection.h(36): message : see declaration of 'Mantid::Algorithms::DiscusData1D'
```
`DiscusData1D` is a struct with x and y and no histogram method that I can see...
Perhaps it's slightly misleading for `DiscusData2D.hitogram()` to produce a `DiscusData1D` struct - or maybe the `DiscusData1D` needs an xmode attribute?
Or maybe I'm not building it right as the jenkins tests seem OK...

Reply:
Yes this was a genuine problem. I checked and a local Release build worked with this in place so the compiler must disregard these lines when doing a Release build (which is what Jenkins does). I'll fix